Note that the architecture forbids downgrade (#1164)
ietf-tapswg/api-drafts: draft-ietf-taps-impl.md

## Considerations for Candidate Gathering

> The Security Considerations of the TAPS Architecture {{I-D.ietf-taps-arch}} forbids gathering and racing with protocol stacks that do not have equivalent security properties. Therefore, implementations need to avoid downgrade attacks that allow network interference to cause the implementation to select less secure, or entirely insecure, combinations of paths and protocols.

Reviewer: ```suggestion
The Security Considerations of the Transport Services Architecture {{I-D.ietf-taps-arch}} forbids gathering and racing with protocol stacks that do not have equivalent security properties. Therefore, implementations need to avoid downgrade attacks that allow network interference to cause the implementation to select less secure, or entirely insecure, combinations of paths and protocols.
```

> As described in the Security Considerations of {{I-D.ietf-taps-arch}}, TAPS forbids applications from specifying protocol stacks with inequivalent security properties for gathering and racing. Implementations must avoid downgrade attacks that allow network interference to cause the implementation to select less secure, or entirely insecure, combinations of paths and protocols.

Reviewer 2: Can we try: "The Security Considerations of the TAPS Architecture {{I-D.ietf-taps-arch}} forbids gathering and racing with protocol stacks that do not have equivalent security properties. Therefore, implementations need to..."

Author: yep that's better

> As described in the Security Considerations of {{I-D.ietf-taps-arch}}, TAPS forbids applications from specifying protocol stacks with inequivalent security properties for gathering and racing. Implementations must avoid downgrade attacks that allow network interference to cause the implementation to select less secure, or entirely insecure, combinations of paths and protocols.

Author: ```suggestion
The Security Considerations of the TAPS Architecture {{I-D.ietf-taps-arch}} forbids gathering and racing with protocol stacks that do not have equivalent security properties. Therefore, implementations need to avoid downgrade attacks that allow network interference to cause the implementation to select less secure, or entirely insecure, combinations of paths and protocols.
```
